src/phasorpy/phasor.py

```diff
@@ -0,0 +1,65 @@
+"""
+============================================
+Phasor functions, module: "phasorpy.phasor"
+============================================
+
+This module provides a set of function to 
+compute phasor transform based on the fast 
+fourier transform algorithm from numpy.fft. 
+There are general tools common to FLIM and 
+HSI data, that helps when doing phasor 
+analysis. 
+
+"""
+
+from __future__ import annotations
+
+__all__ = ['phasor']
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ._typing import Any, NDArray
+
+import numpy as np
+
+
+def phasor(
+    imstack: NDArray[Any], harmonic: int = 1
+) -> tuple[NDArray[Any], NDArray[Any], NDArray[Any]]:
+    """
+    Computes the phasor transform using the
+    fast fourier transform algorithm from numpy.fft
+
+    Parameters
+    ----------
+    imstack : array_like
+        description: Input array, which can be complex.
+            dims = ZXY. Where z is along axis 0.
+            Generally an image stack.
+    harmonic : int, optional
+        description:, by default 1, correspond to the nth
+            harmonic.
+    Returns
+    -------
+        dc: nd array or tuple.
+            Contains the average of the 0-axis
+        g: nd array or tuple.
+            Contains the real part of the fft transform.
+        s: nd array or tuple.
+            Contains the imaginary part of the fft transform.
+
+    """
```

Comment:
The function should accept array-like, not just image stacks. Ideally it should work with any kind and any dimensional array input, including list, tuples, numpy arrays, xarrays, and objects implementing the buffer protocol.

As discussed in our last call, the ``axis`` over which FFT is performed should be an argument. It could be determined from the xarray metadata.

How about renaming ``imstack`` to ``data`` or ``signal`` and make it an positional-only argument? ``axis`` and ``harmonic`` should be keyword-only arguments?

Why/when can the input array be complex?

Document that the input signal must be ordered and evenly-spaced/uniform. Some frequency-domain or spectral signals might not fulfill these requirements.

fft -> FFT

fourier -> Fourier

Non-numpy FFT functions may be more efficient or more general, e.g. [scipy.fft.fft](https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.fft.html), [mkl-fft](https://pypi.org/project/mkl-fft/), or [CuPy](https://docs.cupy.dev/en/stable/user_guide/fft.html). Maybe, allow the user to pass a FFT function?

FFTs may consume more memory than necessary, computing and storing all harmonics. First harmonics may be more efficiently calculated "manually" by multiplying with sin/cos terms. This could be added later.

Follow numpy and Python docstring standards.